functions/gravityForms/getGfFieldValidationSchema.js

```diff
@@ -1,39 +1,116 @@
+ * Create Yup string schema object.
  *
- * @param {Object} fieldData GravityForm field props.
+ * @author Mike England <mike.england@webdevstudios.com>
+ * @since 2021-01-07
+ * @link https://github.com/jquense/yup#string
  */
-export default function getGfFieldValidationSchema(fieldData) {
-  const validationTypes = {
-    text: 'string'
+class StringSchemaFactory {
+  /**
+   * Constructor.
+   *
+   * Note: wp-graphql-gravity-forms plugin may need to be replaced.
+   *
+   * @param {Object} fieldData from GravityForm GraphQL data Object.
+   * @link https://github.com/harness-software/wp-graphql-gravity-forms
+   */
+  constructor(fieldData) {
+    this.fieldData = fieldData
   }
 
-  const type = validationTypes?.[fieldData?.type]
-
-  if (!type) {
-    return
+  /**
+   * Get Yup schema object with validation requirements.
+   *
+   * Combine multiple schemas into one.
+   *
+   * @author Mike England <mike.england@webdevstudios.com>
+   * @since 2021-01-07
+   * @returns {Object} Combined Yup validationSchema Object.
+   * @link https://github.com/jquense/yup#mixedconcatschema-schema-schema
+   */
+  get schema() {
+    return Yup.string()
+      .concat(this.getMaxLengthSchema())
+      .concat(this.getRequiredSchema())
   }
 
-  const yupType = Yup[type]()
+  /**
+   * Get Yup required field validaion.
+   *
+   * @author Mike England <mike.england@webdevstudios.com>
+   * @since 2021-01-07
+   * @return {Object} Yup validationSchema Object.
+   * @link https://github.com/jquense/yup#stringrequiredmessage-string--function-schema
+   */
+  getRequiredSchema() {
+    if (!this.fieldData?.isRequired) {
+      return
+    }
+
+    return Yup.string().required('Required!')
+  }
 
   /**
-   * Conditionally concat Yup validations.
+   * Get Yup max line length validation.
+   *
+   * @author Mike England <mike.england@webdevstudios.com>
+   * @since 2021-01-07
+   * @return {Object} Yup validationSchema Object.
+   * @link https://github.com/jquense/yup#stringmaxlimit-number--ref-message-string--function-schema
    */
-  const validationSchema = yupType
-    .concat(
-      fieldData?.maxLength
-        ? yupType.max(
-            fieldData.maxLength,
-            `Must be ${fieldData.maxLength} characters or less`
-          )
-        : null
+  getMaxLengthSchema() {
+    if (!this.fieldData?.maxLength) {
+      return
+    }
+
+    return Yup.string().max(
+      this.fieldData.maxLength,
+      `Must be ${this.fieldData.maxLength} characters or less`
     )
-    .concat(fieldData?.isRequired ? yupType.required('Required!') : null)
+  }
+}
+
+/**
+ * Match field type with Yup schema object.
+ *
+ * Match GravityForm field to Yup API property.
+ *
+ * @param {Object} fieldData GravityForm field props.
+ * @returns {Object} Schema validation for field.
+ *
+ * @author Mike England <mike.england@webdevstudios.com>
+ * @since 2021-01-07
+ * @link https://github.com/jquense/yup#api
+ */
+function getValidationSchemaType(fieldData) {
```

Comment:
What do you think about renaming this to "getValidationSchema**By**Type" for clarity?

Reply:
I like this name even better, I'll get it updated.

Reply:
Got this updated. I like it.